server/src/main/java/org/elasticsearch/common/document/DocumentField.java

```diff
@@ -135,10 +135,12 @@ public ToXContentFragment getValidValuesWriter() {
         return (builder, params) -> {
             builder.startArray(name);
             for (Object value : values) {
-                // This call doesn't really need to support writing any kind of object, since the values
-                // here are always serializable to xContent. Each value could be a leaf types like a string,
-                // number, or boolean, a list of such values, or a map of such values with string keys.
-                builder.value(value);
+                try {
+                    builder.value(value);
+                } catch (RuntimeException e) {
+                    // if the value cannot be serialized, we catch here and return a placeholder value
```

Comment:
Since we are catching a generic RuntimeException, we don't know if this was an intentional error (`UnsupportedOperationException`) or unintentional (NPE or other type of unexpected error). Should we inspect the error and log it if it is not `UnsupportedOperationException`?

Reply:
We could make the placeholder value more explicit by including the exception message maybe?  I'm not sure how frequent this issue actually is, or how useful an error message would be for something like a script field.

Reply:
I'd vote for logging to help find rare bugs and avoid "swallowing" unexpected errors, but I'm still ramping up on standard practices here and don't have much context on this issue yet :-) 
But this is just optional feedback.

Reply:
Having thought about this, I think I'd like to leave it as it is for now - really the only useful information we could return here would be a stack trace, which isn't really very user friendly.  It shouldn't be too difficult to reproduce a problem if it comes up as we will have the document source.

Reply:
this reminded me of [Strings#toString](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/common/Strings.java#L818), but there we catch `IOException` which is a checked exception. What are the exceptions that can happen in practice? I see that `XContentBuilder#value` may throw `IllegalArgumentException`. We could make the builder print out a more user-friendly error?